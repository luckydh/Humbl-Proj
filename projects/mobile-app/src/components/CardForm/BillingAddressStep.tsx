import React from "react";
import { useForm } from "react-hook-form";
import cx from "classnames";
import { Input } from "../Input/Input";
import { Button } from "../Button/Button";
import { Message } from "../Message/Message";
import { useTranslation } from "react-i18next";
import { CountrySelect } from "components/CountrySelect/CountrySelect";
import { GetRegexByPostalCode } from "utils/PostalCode";
import { Select } from "components/Select/Select";
import getStateOptions from "utils/stateOptions";
import { GetCountryByString } from "utils/CountriesMapping";

export type BillingAddressInfo = {
  country: string;
  city: string;
  state: string;
  postalCode: string;
  addressLine1: string;
  addressLine2: string;
};

interface Props {
  initialData?: BillingAddressInfo;
  onComplete: (data: BillingAddressInfo) => Promise<void>;
  submitButton?: JSX.Element;
  variant?: "fill" | "normal";
  ariaLabel?: string;
}

const BillingAddressStep: React.FC<Props> = ({ initialData, onComplete, submitButton, variant, ariaLabel }) => {
  const { t } = useTranslation();
  const { register, handleSubmit, errors, watch, getValues } = useForm<BillingAddressInfo>({
    mode: "onBlur",
    defaultValues: initialData,
  });

  watch("country");
  watch("state");
  const selectedCountry = (getValues("country") as string) || "";
  const stateOptions = getStateOptions(GetCountryByString(selectedCountry));
  const currentCountryHasNoStates = stateOptions.length === 0;

  const submitHandler = () => {
    const values = getValues();
    // currently the form does not return any key for state if there are no states.
    // the backend requires at least an empty string.
    if (currentCountryHasNoStates) {
      values.state = "";
    }
    onComplete(values);
  };

  return (
    <form
      onSubmit={handleSubmit(submitHandler)}
      className={cx("flex flex-col justify-center h-full", { "w-full": variant === "fill" })}>
      <div className="mb-4">
        <CountrySelect
          label={t("page-billing-address-step.label.country")}
          name="country"
          register={register({
            required: { message: t("add-card-billing-address-country-required"), value: true },
          })}
          ariaLabel={ariaLabel}
        />
        {errors.country?.message && (
          <div className="mt-2">
            <Message
              variant="error"
              ariaLabel={ariaLabel ? `${ariaLabel}_COUNTRY_SELECT_ERROR` : "COUNTRY_SELECT_ERROR"}>
              {errors.country?.message}
            </Message>
          </div>
        )}
      </div>
      <div className="mb-4">
        <Input
          label={t("page-billing-address-step.label.billing-address")}
          name="addressLine1"
          placeholder="Address Line 1"
          className={errors.addressLine1?.message ? "" : "mb-4"}
          register={register({ required: `${t("billing-address-step.address1-error")}` })}
          ariaLabel={ariaLabel ? `${ariaLabel}_ADDRESS_LINE1"` : "ADDRESS_LINE1"}
        />
        {errors.addressLine1?.message && (
          <div className="my-2">
            <Message variant="error" ariaLabel={ariaLabel ? `${ariaLabel}_ADDRESS_LINE1_ERROR` : "ADDRESS_LINE1_ERROR"}>
              {errors.addressLine1?.message}
            </Message>
          </div>
        )}
        <Input
          name="addressLine2"
          placeholder={t("page-billing-address-step.placeholder.address-line-2")}
          register={register}
          ariaLabel={ariaLabel ? `${ariaLabel}_ADDRESS_LINE2` : "ADDRESS_LINE2"}
        />
      </div>
      <div className="mb-4">
        <Input
          label={t("page-billing-address-step.label.city")}
          name="city"
          placeholder={t("page-billing-address-step.label.city")}
          register={register({ required: `${t("billing-address-step.city-error")}` })}
          ariaLabel={ariaLabel ? `${ariaLabel}_CITY_INPUT` : "CITY_INPUT"}
        />
        {errors.city?.message && (
          <div className="mt-2">
            <Message variant="error" ariaLabel={ariaLabel ? `${ariaLabel}_CITY_INPUT_ERROR` : "CITY_INPUT_ERROR"}>
              {errors.city?.message}
            </Message>
          </div>
        )}

        {!currentCountryHasNoStates && (
          <div className="flex-1 mt-4">
            <label className="flex opacity-80 ml-2 mb-1 uppercase text-xs" htmlFor="state">
              {t("page-billing-address-step.label.state-region")}
            </label>
            <div className="flex flex-col">
              <Select
                options={stateOptions}
                name="state"
                register={register({ required: `${t("billing-address-step.state-select-error")}` })}
                placeholder={t("page-billing-address-step.label.state-region")}
              />
              {errors.state?.message && (
                <div className="mt-2">
                  <Message variant="error">{errors?.state?.message}</Message>
                </div>
              )}
            </div>
          </div>
        )}
      </div>
      <div className="mb-8">
        <div>
          <Input
            label={t("page-billing-address-step.label.postal-code")}
            name="postalCode"
            register={
              register({
                required: {
                  value: true,
                  message: t("page-billing-address-step.error.postal-code.required"),
                },
                pattern: {
                  value: watch("country") && GetRegexByPostalCode(watch("country")).Regex,
                  message: t("page-billing-address-step.error.postal-code.invalid"),
                },
              } as any) as any
            }
            placeholder={t("page-billing-address-step.label.postal-code")}
            ariaLabel={ariaLabel ? `${ariaLabel}_POST_CODE` : "POST_CODE"}
          />
          {errors.postalCode?.message && (
            <div className="mt-2">
              <Message variant="error" ariaLabel={ariaLabel ? `${ariaLabel}_POST_CODE_ERROR` : "POST_CODE_ERROR"}>
                {errors.postalCode?.message}
              </Message>
            </div>
          )}
        </div>
        <div className="flex-1" />
      </div>
      <div className="flex justify-center mt-auto">
        {submitButton || (
          <Button type="submit">
            {!initialData
              ? t("page-credit-card-info-step.button.save-changes")
              : t("page-billing-address-step.button.add-billing-address")}
          </Button>
        )}
      </div>
    </form>
  );
};

export default BillingAddressStep;
