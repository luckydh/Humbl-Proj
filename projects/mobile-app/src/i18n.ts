import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
