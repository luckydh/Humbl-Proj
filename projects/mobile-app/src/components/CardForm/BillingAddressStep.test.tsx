import React from "react";
import { StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { CardWithAddress } from "__fixtures__/paymentMethods";

// Disable suspense on menu drawer
import i18n from "../../i18n";
import BillingAddressStep from "./BillingAddressStep";

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}
it("renders Credit Card Billing Address Step form without data", () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <BillingAddressStep onComplete={() => new Promise(() => {})} />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Credit Card Billing Address Step form with initial data", () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <BillingAddressStep
          initialData={{
            country: CardWithAddress.country,
            city: CardWithAddress.cityAddress,
            state: CardWithAddress.region,
            postalCode: CardWithAddress.PostalAddress,
            addressLine1: CardWithAddress.streetAddress,
            addressLine2: CardWithAddress.streetAdditional,
          }}
          onComplete={() => new Promise(() => {})}
        />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
