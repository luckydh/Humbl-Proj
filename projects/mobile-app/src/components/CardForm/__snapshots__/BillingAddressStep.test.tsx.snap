// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Credit Card Billing Address Step form with initial data 1`] = `
<form
  className="flex flex-col justify-center h-full"
  onSubmit={[Function]}
>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.country
        </label>
      </div>
      <select
        className="py-2 px-4 w-full rounded-lg bg-blue border-white border-2 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:outline-none focus:ring-0 focus:placeholder-transparent"
        name="country"
        style={
          Object {
            "fontSize": 20,
            "height": 52,
          }
        }
      >
        <option
          value=""
        >
          merchant-create-page.option.select-a-country
        </option>
        <option
          aria-label="AU"
          value="au"
        >
          Australia
        </option>
        <option
          aria-label="CA"
          value="ca"
        >
          Canada
        </option>
        <option
          aria-label="MX"
          value="mx"
        >
          Mexico
        </option>
        <option
          aria-label="NZ"
          value="nz"
        >
          New Zealand
        </option>
        <option
          aria-label="SG"
          value="sg"
        >
          Singapore
        </option>
        <option
          aria-label="US"
          value="us"
        >
          United States of America
        </option>
      </select>
    </div>
  </div>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          aria-label="ADDRESS_LINE1_LABEL"
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.billing-address
        </label>
      </div>
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none mb-4"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="addressLine1"
          placeholder="Address Line 1"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mb-2"
    >
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="addressLine2"
          placeholder="page-billing-address-step.placeholder.address-line-2"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
  </div>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          aria-label="CITY_INPUT_LABEL"
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.city
        </label>
      </div>
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="city"
          placeholder="page-billing-address-step.label.city"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="flex-1 mt-4"
    >
      <label
        className="flex opacity-80 ml-2 mb-1 uppercase text-xs"
        htmlFor="state"
      >
        page-billing-address-step.label.state-region
      </label>
      <div
        className="flex flex-col"
      >
        <select
          className="py-2 px-4 w-full rounded-lg bg-blue border-white border-2 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:outline-none focus:ring-0 focus:placeholder-transparent"
          defaultValue=""
          name="state"
          style={
            Object {
              "fontSize": 20,
              "height": 52,
            }
          }
        >
          <option
            disabled={true}
            value=""
          >
            page-billing-address-step.label.state-region
          </option>
          <option
            value="AL"
          >
            Alabama
          </option>
          <option
            value="AK"
          >
            Alaska
          </option>
          <option
            value="AS"
          >
            American Samoa
          </option>
          <option
            value="AZ"
          >
            Arizona
          </option>
          <option
            value="AR"
          >
            Arkansas
          </option>
          <option
            value="CA"
          >
            California
          </option>
          <option
            value="CO"
          >
            Colorado
          </option>
          <option
            value="CT"
          >
            Connecticut
          </option>
          <option
            value="DE"
          >
            Delaware
          </option>
          <option
            value="DC"
          >
            District Of Columbia
          </option>
          <option
            value="FL"
          >
            Florida
          </option>
          <option
            value="GA"
          >
            Georgia
          </option>
          <option
            value="GU"
          >
            Guam
          </option>
          <option
            value="HI"
          >
            Hawaii
          </option>
          <option
            value="ID"
          >
            Idaho
          </option>
          <option
            value="IL"
          >
            Illinois
          </option>
          <option
            value="IN"
          >
            Indiana
          </option>
          <option
            value="IA"
          >
            Iowa
          </option>
          <option
            value="KS"
          >
            Kansas
          </option>
          <option
            value="KY"
          >
            Kentucky
          </option>
          <option
            value="LA"
          >
            Louisiana
          </option>
          <option
            value="ME"
          >
            Maine
          </option>
          <option
            value="MD"
          >
            Maryland
          </option>
          <option
            value="MA"
          >
            Massachusetts
          </option>
          <option
            value="MI"
          >
            Michigan
          </option>
          <option
            value="MN"
          >
            Minnesota
          </option>
          <option
            value="MS"
          >
            Mississippi
          </option>
          <option
            value="MO"
          >
            Missouri
          </option>
          <option
            value="MT"
          >
            Montana
          </option>
          <option
            value="NE"
          >
            Nebraska
          </option>
          <option
            value="NV"
          >
            Nevada
          </option>
          <option
            value="NH"
          >
            New Hampshire
          </option>
          <option
            value="NJ"
          >
            New Jersey
          </option>
          <option
            value="NM"
          >
            New Mexico
          </option>
          <option
            value="NY"
          >
            New York
          </option>
          <option
            value="NC"
          >
            North Carolina
          </option>
          <option
            value="ND"
          >
            North Dakota
          </option>
          <option
            value="OH"
          >
            Ohio
          </option>
          <option
            value="OK"
          >
            Oklahoma
          </option>
          <option
            value="OR"
          >
            Oregon
          </option>
          <option
            value="PA"
          >
            Pennsylvania
          </option>
          <option
            value="PR"
          >
            Puerto Rico
          </option>
          <option
            value="RI"
          >
            Rhode Island
          </option>
          <option
            value="SC"
          >
            South Carolina
          </option>
          <option
            value="SD"
          >
            South Dakota
          </option>
          <option
            value="TN"
          >
            Tennessee
          </option>
          <option
            value="TX"
          >
            Texas
          </option>
          <option
            value="UT"
          >
            Utah
          </option>
          <option
            value="VT"
          >
            Vermont
          </option>
          <option
            value="VI"
          >
            Virgin Islands
          </option>
          <option
            value="VA"
          >
            Virginia
          </option>
          <option
            value="WA"
          >
            Washington
          </option>
          <option
            value="WV"
          >
            West Virginia
          </option>
          <option
            value="WI"
          >
            Wisconsin
          </option>
          <option
            value="WY"
          >
            Wyoming
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mb-8"
  >
    <div>
      <div
        className="mb-2"
      >
        <div
          className="ml-2"
        >
          <label
            aria-label="POST_CODE_LABEL"
            className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
          >
            page-billing-address-step.label.postal-code
          </label>
        </div>
        <div
          className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
        >
          <input
            className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
            name="postalCode"
            placeholder="page-billing-address-step.label.postal-code"
            style={
              Object {
                "fontSize": 20,
                "height": 48,
              }
            }
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="flex-1"
    />
  </div>
  <div
    className="flex justify-center mt-auto"
  >
    <button
      className="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none"
      onClick={[Function]}
      style={
        Object {
          "minWidth": 160,
        }
      }
      type="submit"
    >
      page-billing-address-step.button.add-billing-address
    </button>
  </div>
</form>
`;

exports[`renders Credit Card Billing Address Step form without data 1`] = `
<form
  className="flex flex-col justify-center h-full"
  onSubmit={[Function]}
>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.country
        </label>
      </div>
      <select
        className="py-2 px-4 w-full rounded-lg bg-blue border-white border-2 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:outline-none focus:ring-0 focus:placeholder-transparent"
        name="country"
        style={
          Object {
            "fontSize": 20,
            "height": 52,
          }
        }
      >
        <option
          value=""
        >
          merchant-create-page.option.select-a-country
        </option>
        <option
          aria-label="AU"
          value="au"
        >
          Australia
        </option>
        <option
          aria-label="CA"
          value="ca"
        >
          Canada
        </option>
        <option
          aria-label="MX"
          value="mx"
        >
          Mexico
        </option>
        <option
          aria-label="NZ"
          value="nz"
        >
          New Zealand
        </option>
        <option
          aria-label="SG"
          value="sg"
        >
          Singapore
        </option>
        <option
          aria-label="US"
          value="us"
        >
          United States of America
        </option>
      </select>
    </div>
  </div>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          aria-label="ADDRESS_LINE1_LABEL"
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.billing-address
        </label>
      </div>
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none mb-4"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="addressLine1"
          placeholder="Address Line 1"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mb-2"
    >
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="addressLine2"
          placeholder="page-billing-address-step.placeholder.address-line-2"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
  </div>
  <div
    className="mb-4"
  >
    <div
      className="mb-2"
    >
      <div
        className="ml-2"
      >
        <label
          aria-label="CITY_INPUT_LABEL"
          className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
        >
          page-billing-address-step.label.city
        </label>
      </div>
      <div
        className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
      >
        <input
          className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
          name="city"
          placeholder="page-billing-address-step.label.city"
          style={
            Object {
              "fontSize": 20,
              "height": 48,
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="flex-1 mt-4"
    >
      <label
        className="flex opacity-80 ml-2 mb-1 uppercase text-xs"
        htmlFor="state"
      >
        page-billing-address-step.label.state-region
      </label>
      <div
        className="flex flex-col"
      >
        <select
          className="py-2 px-4 w-full rounded-lg bg-blue border-white border-2 outline-none text-white placeholder-white-faded text-xl block focus:ring-indigo-500 focus:border-white focus:outline-none focus:ring-0 focus:placeholder-transparent"
          defaultValue=""
          name="state"
          style={
            Object {
              "fontSize": 20,
              "height": 52,
            }
          }
        >
          <option
            disabled={true}
            value=""
          >
            page-billing-address-step.label.state-region
          </option>
          <option
            value="AL"
          >
            Alabama
          </option>
          <option
            value="AK"
          >
            Alaska
          </option>
          <option
            value="AS"
          >
            American Samoa
          </option>
          <option
            value="AZ"
          >
            Arizona
          </option>
          <option
            value="AR"
          >
            Arkansas
          </option>
          <option
            value="CA"
          >
            California
          </option>
          <option
            value="CO"
          >
            Colorado
          </option>
          <option
            value="CT"
          >
            Connecticut
          </option>
          <option
            value="DE"
          >
            Delaware
          </option>
          <option
            value="DC"
          >
            District Of Columbia
          </option>
          <option
            value="FL"
          >
            Florida
          </option>
          <option
            value="GA"
          >
            Georgia
          </option>
          <option
            value="GU"
          >
            Guam
          </option>
          <option
            value="HI"
          >
            Hawaii
          </option>
          <option
            value="ID"
          >
            Idaho
          </option>
          <option
            value="IL"
          >
            Illinois
          </option>
          <option
            value="IN"
          >
            Indiana
          </option>
          <option
            value="IA"
          >
            Iowa
          </option>
          <option
            value="KS"
          >
            Kansas
          </option>
          <option
            value="KY"
          >
            Kentucky
          </option>
          <option
            value="LA"
          >
            Louisiana
          </option>
          <option
            value="ME"
          >
            Maine
          </option>
          <option
            value="MD"
          >
            Maryland
          </option>
          <option
            value="MA"
          >
            Massachusetts
          </option>
          <option
            value="MI"
          >
            Michigan
          </option>
          <option
            value="MN"
          >
            Minnesota
          </option>
          <option
            value="MS"
          >
            Mississippi
          </option>
          <option
            value="MO"
          >
            Missouri
          </option>
          <option
            value="MT"
          >
            Montana
          </option>
          <option
            value="NE"
          >
            Nebraska
          </option>
          <option
            value="NV"
          >
            Nevada
          </option>
          <option
            value="NH"
          >
            New Hampshire
          </option>
          <option
            value="NJ"
          >
            New Jersey
          </option>
          <option
            value="NM"
          >
            New Mexico
          </option>
          <option
            value="NY"
          >
            New York
          </option>
          <option
            value="NC"
          >
            North Carolina
          </option>
          <option
            value="ND"
          >
            North Dakota
          </option>
          <option
            value="OH"
          >
            Ohio
          </option>
          <option
            value="OK"
          >
            Oklahoma
          </option>
          <option
            value="OR"
          >
            Oregon
          </option>
          <option
            value="PA"
          >
            Pennsylvania
          </option>
          <option
            value="PR"
          >
            Puerto Rico
          </option>
          <option
            value="RI"
          >
            Rhode Island
          </option>
          <option
            value="SC"
          >
            South Carolina
          </option>
          <option
            value="SD"
          >
            South Dakota
          </option>
          <option
            value="TN"
          >
            Tennessee
          </option>
          <option
            value="TX"
          >
            Texas
          </option>
          <option
            value="UT"
          >
            Utah
          </option>
          <option
            value="VT"
          >
            Vermont
          </option>
          <option
            value="VI"
          >
            Virgin Islands
          </option>
          <option
            value="VA"
          >
            Virginia
          </option>
          <option
            value="WA"
          >
            Washington
          </option>
          <option
            value="WV"
          >
            West Virginia
          </option>
          <option
            value="WI"
          >
            Wisconsin
          </option>
          <option
            value="WY"
          >
            Wyoming
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mb-8"
  >
    <div>
      <div
        className="mb-2"
      >
        <div
          className="ml-2"
        >
          <label
            aria-label="POST_CODE_LABEL"
            className="block text-xs uppercase text-white opacity-80 font-medium mb-1"
          >
            page-billing-address-step.label.postal-code
          </label>
        </div>
        <div
          className="py-0 w-full flex items-center rounded-lg select-none bg-blue border-white border-2 border-solid outline-none"
        >
          <input
            className="text-white placeholder-white-faded w-full rounded-lg bg-transparent text-md sm:text-xl outline-none border-none focus:border-transparent focus:outline-none focus:ring-0 focus:placeholder-transparent"
            name="postalCode"
            placeholder="page-billing-address-step.label.postal-code"
            style={
              Object {
                "fontSize": 20,
                "height": 48,
              }
            }
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="flex-1"
    />
  </div>
  <div
    className="flex justify-center mt-auto"
  >
    <button
      className="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none"
      onClick={[Function]}
      style={
        Object {
          "minWidth": 160,
        }
      }
      type="submit"
    >
      page-credit-card-info-step.button.save-changes
    </button>
  </div>
</form>
`;
